crates/arroyo-connectors/src/nats/source/mod.rs

```diff
@@ -0,0 +1,548 @@
+        );
+        info!(
+            "Stream - number of bytes contained: {}",
+            &stream_info.state.bytes
+        );
+        info!(
+            "Stream - number of consumers: {}",
+            &stream_info.state.consumer_count
+        );
+
+        stream
+    }
+
+    async fn create_nats_consumer(
+        &mut self,
+        stream: &async_nats::jetstream::stream::Stream,
+        sequence_number: u64,
+        ctx: &mut ArrowContext,
+    ) -> consumer::Consumer<consumer::pull::Config> {
+        match sequence_number {
+            1 => info!(
+                ">> No state found for NATS, starting from sequence number #{}",
+                &sequence_number
+            ),
+            _ => info!(
+                ">> Found state for NATS, starting from sequence number #{}",
+                &sequence_number
+            ),
+        };
+
+        let deliver_policy = {
+            if sequence_number == 1 {
+                consumer::DeliverPolicy::All
+            } else {
+                consumer::DeliverPolicy::ByStartSequence {
+                    start_sequence: sequence_number,
+                }
+            }
+        };
+
+        let consumer_name = format!(
+            "{}-{}",
+            match &self.source_type {
+                SourceType::Jetstream { stream, .. } => {
+                    stream
+                }
+                SourceType::Core { subject, .. } => {
+                    subject
+                }
+            },
+            &ctx.task_info.operator_id.replace("operator_", "")
+        );
+
+        let consumer_config = match &self.source_type {
+            SourceType::Jetstream {
+                ack_policy,
+                replay_policy,
+                ack_wait,
+                description,
+                filter_subjects,
+                rate_limit,
+                sample_frequency,
+                num_replicas,
+                inactive_threshold,
+                max_ack_pending,
+                max_deliver,
+                max_waiting,
+                max_batch,
+                max_bytes,
+                max_expires,
+                ..
+            } => consumer::pull::Config {
+                name: Some(consumer_name.clone()),
+                ack_policy: match ack_policy {
+                    AcknowledgmentPolicy::Explicit => consumer::AckPolicy::Explicit,
+                    AcknowledgmentPolicy::All => consumer::AckPolicy::All,
+                    AcknowledgmentPolicy::None => consumer::AckPolicy::None,
+                },
+                replay_policy: match replay_policy {
+                    ReplayPolicy::Original => consumer::ReplayPolicy::Original,
+                    ReplayPolicy::Instant => consumer::ReplayPolicy::Instant,
+                },
+                ack_wait: Duration::from_secs(ack_wait.clone() as u64),
+                description: description.clone(),
+                filter_subjects: filter_subjects
```

Comment:
It appears that this doesn't work properly if it's not set (i.e., `""`). 

I get 

> Something went wrong instantiating the NATS consumer.: Error { kind: JetStream(Error { code: 400, err_code: ErrorCode(10139), description: Some("consumer filter in FilterSubjects cannot be empty") }), source: None } panic.file="crates/arroyo-connectors/src/nats/source/mod.rs" panic.line=281 panic.column=14

Reply:
I think that's fixed, with single quotes at least you should be able to pass an empty string and define it as comma-separated list of strings